x-pack/plugins/reporting/public/plugin.ts

```diff
@@ -236,17 +237,19 @@ export class ReportingPublicPlugin
           })
         );
 
-        share.register(
-          reportingScreenshotShareProvider({
-            apiClient,
-            toasts,
-            uiSettings,
-            license,
-            application,
-            usesUiCapabilities,
-            theme: core.theme,
-          })
-        );
+        if (!this.config.export_types.pdf.enabled || !this.config.export_types.png.enabled) {
```

Comment:
Something is odd with the schema. Before registering the screenshot share provider, we should check if the PDF/PNG types are **enabled**. 🤔

Reply:
I think this suggestion will help: https://github.com/elastic/kibana/pull/162358/files#r1276868544

The final defaultValue outside of serverless should be `true`